handler/handler.go

```diff
@@ -39,78 +33,133 @@ func NewBatchHandler(esClient *dpelasticsearch.Client) *BatchHandler {
 func (batchHandler BatchHandler) Handle(ctx context.Context, url string, events []*models.SearchDataImportModel) error {
 	log.Info(ctx, "events handler called")
 
-	//no events receeived. Nothing more to do.
+	//no events received. Nothing more to do.
 	if len(events) == 0 {
 		log.Info(ctx, "there are no events to handle")
 		return nil
 	}
-	log.Info(ctx, "Events received ", log.Data{
-		"events received": len(events),
-	})
 
-	// This will block if we've reached our concurrency limit (sem buffer size)
-	err := batchHandler.SendToES(ctx, url, events)
+	err := batchHandler.sendToES(ctx, url, events)
 	if err != nil {
-		log.Fatal(ctx, "failed to send event to Elastic Search", err)
+		log.Error(ctx, "failed to send event to Elastic Search", err)
 		return err
 	}
 
 	log.Info(ctx, "event successfully handled")
 	return nil
 }
 
-func (bh BatchHandler) SendToES(ctx context.Context, esDestURL string, events []*models.SearchDataImportModel) error {
-
-	// Wait on semaphore if we've reached our concurrency limit
-	syncWaitGroup.Add(1)
-	semaphore <- 1
-
-	t := transform.NewTransformer()
-
-	go func() {
+// Preparing the payload and sending bulk events to elastic search.
+func (batchHandler BatchHandler) sendToES(ctx context.Context, esDestURL string, events []*models.SearchDataImportModel) error {
 
-		defer func() {
-			<-semaphore
-			syncWaitGroup.Done()
-		}()
+	log.Info(ctx, "bulk events into ES starts")
+	target := len(events)
+	documentList := make(map[string]models.SearchDataImportModel)
 
-		log.Info(ctx, "go routine for inserting into ES starts")
-
-		var bulk []byte
+	var bulkcreate []byte
+	for _, event := range events {
+		if event.Title == "" {
+			log.Info(ctx, "No title for inbound event, no transformation possible")
+			continue // break here
+		}
 
-		i := 0
-		for i < len(events) {
+		documentList[event.Title] = *event
+		createBulkRequestBody, err := prepareEventForBulkRequestBody(ctx, event, "create")
+		if err != nil {
+			log.Error(ctx, "error in preparing the bulk for create", err, log.Data{
+				"event": *event,
+			})
+		}
```

Comment:
Probably want to `continue` here otherwise bulkcreate variable will likely panic